DirectProgramming/C++SYCL_FPGA/Tutorials/Tools/use_library/part2_lib_func_call/src/lib_rtl.hpp

```diff
@@ -4,6 +4,9 @@
 // SPDX-License-Identifier: MIT
 // =============================================================
 #include <sycl/sycl.hpp>
+#include <sycl/ext/intel/ac_types/ac_int.hpp>
+using MyInt27 = ac_int<27, false>;
+using MyInt54 = ac_int<54, false>;
```

Comment:
better to include `stdint.h` and use `uint32_t` and `uint64_t` for the inputs and output. this will silence all the warnings:

```
/nfs/site/disks/swuser_work_whitepau/oneAPI_Examples/oneAPI-samples.junys-use_library/DirectProgramming/C++SYCL_FPGA/Tutorials/Tools/use_library/part2_lib_func_call/src/lib_rtl_model.cpp:11:34: warning: 'RtlDSPm27x27u' has C-linkage specified, but returns incomplete type 'MyInt54' (aka 'ac_int<54, false>') which could be incompatible with C [-Wreturn-type-c-linkage]
   11 | SYCL_EXTERNAL extern "C" MyInt54 RtlDSPm27x27u (MyInt27 x, MyInt27 y) {
      |                                  ^
1 warning generated.
/nfs/site/disks/swuser_work_whitepau/oneAPI_Examples/oneAPI-samples.junys-use_library/DirectProgramming/C++SYCL_FPGA/Tutorials/Tools/use_library/part2_lib_func_call/src/lib_rtl_model.cpp:11:34: warning: 'RtlDSPm27x27u' has C-linkage specified, but returns incomplete type 'MyInt54' (aka 'ac_int<54, false>') which could be incompatible with C [-Wreturn-type-c-linkage]
   11 | SYCL_EXTERNAL extern "C" MyInt54 RtlDSPm27x27u (MyInt27 x, MyInt27 y) {
      |                                  ^
1 warning generated.
```

Reply:
even when the RTL is expecting 27-bit inputs?

Reply:
The RTL should trim out the top 5 bits. something like this:

```verilog
wire [31:0] my32bit;
wire [26:0] my27bit;

assign my27bit = my32bit[26:0];
```

Please file a HSD ES bug about that warning.

Reply:
I come across [1806960318](https://hsdes.intel.com/appstore/article/#/1806960318) when trying to file new HSD case.
Do you see the warning as the same as this case?
Is there still need to open new case or we could follow up on this?

Reply:
I have added a note in readme to inform user to safely ignore this warning